cpp/src/gandiva/filter.cc

```diff
@@ -102,14 +103,30 @@ Status Filter::Make(SchemaPtr schema, ConditionPtr condition,
   ARROW_RETURN_IF(configuration == nullptr,
                   Status::Invalid("Configuration cannot be null"));
 
-  static Cache<FilterCacheKey, std::shared_ptr<Filter>> cache;
-  FilterCacheKey cache_key(schema, configuration, *(condition.get()));
-  auto cachedFilter = cache.GetModule(cache_key);
-  if (cachedFilter != nullptr) {
-    *filter = cachedFilter;
-    return Status::OK();
+  std::shared_ptr<Cache<BaseCacheKey, std::shared_ptr<llvm::MemoryBuffer>>> shared_cache = LLVMGenerator::GetCache();
+
+  FilterCacheKey filter_key(schema, configuration, *(condition.get()));
```

Comment:
I think you can set `*(condition.get())` to a variable and pass to the function, to make clear that we are retrieving a value from a pointer.

The current way is hard to identify it

Reply:
I did put it on a variable, as you pointed out. But this way above is from the original code.